framework/include/base/MooseError.h

```diff
@@ -34,33 +35,27 @@
 #define MOOSE_ABORT do { std::abort(); } while (0)
 #endif
 
-/**
- * MOOSE wrapped versions of useful libMesh macros (see libmesh_common.h)
- */
-#define mooseError(msg)                                                             \
-  do                                                                                \
-  {                                                                                 \
-    std::ostringstream _error_oss_;                                                 \
-    _error_oss_ << "\n\n"                                                           \
-                << COLOR_RED                                                        \
-                << "\n\n*** ERROR ***\n"                                            \
-                << msg                                                              \
-                << COLOR_DEFAULT                                                    \
-                << "\n\n";                                                          \
-    if (Moose::_throw_on_error)                                                     \
-      throw std::runtime_error(_error_oss_.str());                                  \
-    else                                                                            \
-    {                                                                               \
-      Moose::err << _error_oss_.str() << std::flush;                                \
-      if (libMesh::global_n_processors() == 1)                                      \
-        print_trace();                                                              \
-      else                                                                          \
-        libMesh::write_traceout();                                                  \
-      libmesh_here();                                                               \
-      MOOSE_ABORT;                                                                  \
-    }                                                                               \
-  } while (0)
+std::string mooseMsgFmt(const std::string & msg, const std::string & title, const std::string & color);
+
+[[ noreturn ]] void mooseErrorRaw(std::string msg, const std::string prefix = "");
+
+void mooseStreamAll(std::ostringstream & ss);
 
+template <typename T, typename... Args>
+void mooseStreamAll(std::ostringstream & ss, T val, Args... args)
```

Comment:
I was afraid of that. Maybe we name it something less public or something just so people don't see all of these methods and wonder what to use. Of course your documentation can clearly state the methods to use or you can separate them in the file with nice comments formatting.